### DISCUSSION TITLE
Composer vulnerability report outside github

### DISCUSSION THREAD

**Discussion by icanhazstring** [renovatebot/renovate#16540] [Suggest an Idea]
Some time ago I asked for support for different cve Apis outside dependabot. Which was declined.

Now there are some changes coming to composer with 2.4.

https://github.com/composer/composer/pull/10798

Would it be possible using this to report vulnerabilities?

E.g. Running
```
composer audit
```

This would report current CVEs for installed packages which can be used for non github platforms.

**Comment by PhilipAbed** [comment]
Renovate's main goal is to update dependencies automatically.
we also try to avoid running any package manager commands in renovate.
@rarkins @viceice  wdyt?

**Reply by icanhazstring** [reply]
But you do run `composer update` tho? Otherwise the lock file would not be updated.

Adding `composer audit` would totally improve renovate for non github users.

**Reply by PhilipAbed** [reply]
only in Artifact/post update we run commands to update the lockfile , basically for lockfile maintenance, if we can do without it would be better too, the less package managers that we depend on which are installed on the environment the less problems we will have.

**Reply by icanhazstring** [reply]
True. Another thing would be to use the Packagist cve api to parse CVEs. But that was declined in the past.

So the question would be to just use this single command, if possible. (platform matches, manager matches and version)

**Reply by viceice** [reply]
we are planning to use a new separate solution. @JamieMagee has an open PR to implement security updates outside of GitHub platform

**Reply by JamieMagee** [reply]
The project I'm basing my solution on, OSV[^1], would be able to provide composer security alerts via the GitHub Security Advisories which supports composer.

[^1]: https://github.com/google/osv.dev

**Reply by icanhazstring** [reply]
Sounds nice. Thanks for the heads up.

**Reply by icanhazstring** [reply]
Is there an Issue that I can follow to get updates on that? :)

**Reply by philippe-granet** [reply]
> Is there an Issue that I can follow to get updates on that? :)

Issue: https://github.com/renovatebot/renovate/issues/6562
PR: https://github.com/renovatebot/renovate/pull/15159
